src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,241 @@
+            registrationService.register(first);
+        } catch (RegistrationException e) {
+            return;
+        }
+        fail("RegistrationException should be thrown if age is negative");
+    }
+
+    @Test
+    void register_nullUserValue_notOk() {
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        first.setLogin("seco76nd2");
+        first.setPassword("pasgggsw2");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(first);
+        });
+    }
+
+    @Test
+    void register_validPassword_ok() {
+        first.setAge(19);
+        first.setLogin("firserfdt1");
+        first.setPassword("passdfsw1");
+        User actual;
+        try {
+            actual = registrationService.register(first);
+        } catch (RegistrationException e) {
+            return;
+        }
+        User expected = storageDao.get(first.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_notValidPassword_notOk() {
+        first.setAge(19);
+        first.setLogin("firhtrdst1");
+        first.setPassword("sw1");
+        try {
+            registrationService.register(first);
+        } catch (RegistrationException e) {
+            return;
+        }
+        fail("RegistrationException should be thrown if password is less than 6 characters");
+
+        User user = new User();
+        user.setAge(99);
+        user.setLogin("firktst1");
+        user.setPassword("");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        first.setLogin("sec876ond2");
+        first.setAge(46);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(first);
+        });
+    }
+
+    @Test
+    void register_validLogin_ok() {
+        first.setAge(19);
+        first.setLogin("firlnyst1");
+        first.setPassword("pasoppsw1");
+        User actual;
+        try {
+            actual = registrationService.register(first);
+        } catch (RegistrationException e) {
+            return;
+        }
+        User expected = storageDao.get(first.getLogin());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void register_notValidLogin_notOk() {
+        first.setAge(19);
+        first.setLogin("f1");
+        first.setPassword("ppassew-w1");
+        User user = new User();
+        user.setAge(99);
+        user.setLogin("");
+        user.setPassword("as964sw1");
+        try {
+            registrationService.register(first);
+            registrationService.register(user);
+        } catch (RegistrationException e) {
+            return;
+        }
+        fail("RegistrationException should be thrown if password is less than 6 characters");
```

Comment:
The error message in the `fail` method is incorrect. It should state that a `RegistrationException` should be thrown if the login is less than 6 characters, not the password.